src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs

```diff
@@ -171,7 +171,7 @@ public static Expression ValidateOneArgument(MethodBase method, ExpressionType n
             return arguments;
         }
 
-        public static void RequiresCanRead(Expression expression, string paramName, int idx)
+        public static void RequiresCanRead(Expression expression, string paramName, int idx = -1)
```

Comment:
Would it be worth doing this as an overload? On the one hand, the difference of having one extra argument in the calls is very small, but on the other hand this is called a great many times.